python/bindings/openravepy_robot.cpp

```diff
@@ -482,12 +492,38 @@ std::vector<RobotBase::ConnectedBodyInfoPtr> ExtractConnectedBodyInfoArray(objec
     return vConnectedBodyInfos;
 }
 
+std::vector<RobotBase::GripperInfoPtr> ExtractGripperInfoArray(object pyGripperInfoList)
+{
+    if(IS_PYTHONOBJECT_NONE(pyGripperInfoList)) {
+        return {};
+    }
+    std::vector<RobotBase::GripperInfoPtr> vGripperInfos;
+    try {
+        const size_t arraySize = len(pyGripperInfoList);
+        vGripperInfos.reserve(arraySize);
+        dReal fUnitScale=1;
+        int options = 0;
+        for(size_t iGripperInfo = 0; iGripperInfo < arraySize; iGripperInfo++) {
+            RobotBase::GripperInfoPtr pInfo(new RobotBase::GripperInfo());
+            rapidjson::Document rGripperInfo;
+            toRapidJSONValue(pyGripperInfoList[iGripperInfo], rGripperInfo, rGripperInfo.GetAllocator());
+            pInfo->DeserializeJSON(rGripperInfo, fUnitScale, options);
+            vGripperInfos.push_back(pInfo);
+        }
+    }
+    catch(...) {
+        RAVELOG_WARN("Cannot do ExtractArray for GripperInfo");
```

Comment:
From Rosen: @bucketzxm i really don't like this type of catch all usage...https://github.com/rdiankov/openrave/pull/872